From 4ebab1f8fa8a325db08e5349690eec39c25a3ea2 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:01:00 -0400
Subject: [fs] btrfs: send: squeeze bitfilelds in fs_path

Message-id: <1407179109-13714-174-git-send-email-sandeen@redhat.com>
Patchwork-id: 87018
O-Subject: [RHEL7.1 PATCH 173/422] btrfs: send: squeeze bitfilelds in fs_path
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: David Sterba <dsterba@suse.cz>

commit 1f5a7ff999523e9996befbe03e196eb73370fe36
Author: David Sterba <dsterba@suse.cz>
Date:   Mon Feb 3 19:23:47 2014 +0100

    btrfs: send: squeeze bitfilelds in fs_path

    We know that buf_len is at most PATH_MAX, 4k, and can merge it with the
    reversed member. This saves 3 bytes in favor of inline_buf.

    Signed-off-by: David Sterba <dsterba@suse.cz>
    Signed-off-by: Josef Bacik <jbacik@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/send.c b/fs/btrfs/send.c
index 5b9b82b..4405aae 100644
--- a/fs/btrfs/send.c
+++ b/fs/btrfs/send.c
@@ -53,8 +53,8 @@ struct fs_path {
    char *end;
 
    char *buf;
-   int buf_len;
-   unsigned int reversed:1;
+   unsigned short buf_len:15;
+   unsigned short reversed:1;
    char inline_buf[];
   };
   char pad[PAGE_SIZE];
-- 
1.7.1